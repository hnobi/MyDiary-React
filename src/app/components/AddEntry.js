/* eslint camelcase: ["error", {allow: ["UNSAFE_componentWillMount"]}] */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import newEntry from '../redux/actions/addEntry';
import { getAuthToken } from '../services/AuthToken';
import '../assets/css/main.css';

class AddEntry extends Component {
  state = {
    title: '',
    date: '',
    entry: ''
  };

  componentDidMount() {
    const authToken = getAuthToken();
    const { history } = this.props;
    if (!authToken) {
      history.replace('/login');
    }
  }

  handleOnChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleOnSubmit = (event) => {
    const { history, handleAddEntry } = this.props;
    event.preventDefault();
    handleAddEntry(this.state, history);
  };

  render() {
    const { title, entry, date } = this.state;
    // const { loading, message } = this.props.newEntryData;
    // const statusClassName = loading ? 'loading' : '';
    return (
      <div>
        <nav className="navcolor">
          <div id="logo" className="nav-logo">
            <Link to="/" style={{ display: 'flex' }} className="nav-logo">
              <img src="../assets/img/logo-small2.png" alt="logo" />
              <span> My Diary</span>
            </Link>
          </div>
          <div style={{ margin: 'auto' }} />
          <div className="menu">
            <ul>
              <li>
                <Link to="/entries">View Entries</Link>
              </li>
              <li id="logout">
                <Link to="/logout">Logout</Link>xas
              </li>
            </ul>
          </div>
          <Link to="user_detail.html" id="user">
            <div className="user">
              <img
                src="../assets/img/default-img.png"
                alt="User"
                width="96"
                height="96"
                id="nav-img"
              />
              <span id="nav-name" />
            </div>
          </Link>
        </nav>
        {/* {loading === false && status && <Alert message={message} status={status} />} */}
        {/* {message === 'success' && <Redirect to="/entries" />} */}
        <section>
          <h2 className="heading">Add Entry</h2>
          <div className="container">
            <form id="add-entry" onSubmit={this.handleOnSubmit}>
              <label htmlFor="title">Entry title</label>
              <p id="existed-title">Title already exit in your diary</p>
              <input
                type="text"
                id="title"
                name="title"
                placeholder="Your title..."
                required
                onChange={this.handleOnChange}
                defaultValue={title}
              />
              <label htmlFor="date">Date</label>
              <input
                type="date"
                id="date"
                name="date"
                onChange={this.handleOnChange}
                defaultValue={date}
              />
              <label htmlFor="entry">Entry</label>
              <textarea
                type="text"
                id="entry"
                name="entry"
                placeholder="Your entry..."
                maxLength="500"
                minLength="10"
                required
                onChange={this.handleOnChange}
                defaultValue={entry}
              />
              <input type="submit" value="Submit" />
            </form>
          </div>
        </section>
        <div className="push" />
      </div>
    );
  }
}

AddEntry.propTypes = {
  // saveInput: PropTypes.func,
  handleAddEntry: PropTypes.func,
  // newEntryData: PropTypes.object,
  history: PropTypes.shape({})
};

// const mapStateToProps = state => ({
//   newEntryData: state.newEntryData,
//   statusMessage: state.statusMessage
// });
const mapDispatchToProps = dispatch => ({
  handleAddEntry: (data, history) => dispatch(newEntry(data, history))
});

export default connect(
  null,
  mapDispatchToProps
)(AddEntry);
