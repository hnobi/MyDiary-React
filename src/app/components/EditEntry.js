/* eslint camelcase: ["error", {allow: ["UNSAFE_componentWillMount"]}] */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import fetchEntry from '../redux/actions/viewEntry';
import { updateEntry, saveUpdateInput } from '../redux/actions/updateEntry';
import { getAuthToken } from '../services/AuthToken';
import '../assets/css/main.css';
import Alert from './Alert';

class EditEntry extends Component {
  UNSAFE_componentWillMount() {
    const authToken = getAuthToken();
    const { history } = this.props;
    if (!authToken) {
      history.replace('/login');
    }
  }

  componentDidMount() {
    const { fetchSingleEntry, match } = this.props;
    const { entryId } = match.params;
    const authToken = getAuthToken();
    if (authToken) {
      fetchSingleEntry(entryId);
    }
  }

  render() {
    const {
      updateSingleEntry, updateEntryData, entry, saveInput
    } = this.props;
    const { loading, message, status } = this.props.updateEntryData;
    const { input } = updateEntryData;
    const content = input.entry || entry.entry;
    const title = input.title || entry.title;
    const date = input.date || entry.date;
    return (
      <div>
        <nav className="navcolor">
          <div id="logo" className="nav-logo">
            <Link to="/" style={{ display: 'flex' }} className="nav-logo">
              <img src="../assets/img/logo-small2.png" alt="logo" />
              <span> My Diary</span>
            </Link>
          </div>
          <div style={{ margin: 'auto' }} />
          <div className="menu">
            <ul>
              <li>
                <Link to="/entries">View Entries</Link>
              </li>
              <li id="logout">
                <Link to="/logout">Logout</Link>
              </li>
            </ul>
          </div>
          <Link to="user_detail.html" id="user">
            <div className="user">
              <img
                src="../assets/img/default-img.png"
                alt="User"
                width="96"
                height="96"
                id="nav-img"
              />
              <span id="nav-name" />
            </div>
          </Link>
        </nav>
        {loading === false && <Alert message={message} status={status} />}
        <section>
          <h2 className="heading">Modify Entry</h2>

          <div className="container">
            <form
              id="modify-entry"
              onSubmit={(e) => {
                e.preventDefault();
                updateSingleEntry({ ...input, entryId: entry.id });
              }}
            >
              <label htmlFor="title">Entry title</label>
              <input
                type="text"
                id="title"
                name="title"
                placeholder="Your title..."
                required
                onChange={e => saveInput(e.target.name, e.target.value)}
                defaultValue={title}
              />
              <label htmlFor="date">Date</label>
              <input
                type="date"
                id="date"
                name="date"
                onChange={e => saveInput(e.target.name, e.target.value)}
                defaultValue={date}
              />
              <label htmlFor="entry">Entry</label>
              <textarea
                type="text"
                id="entry"
                name="entry"
                placeholder="Your entry..."
                maxLength="500"
                minLength="10"
                required
                onChange={e => saveInput(e.target.name, e.target.value)}
                defaultValue={content}
              />
              <input type="submit" value="Submit" />
            </form>
          </div>
        </section>
        <div className="push" />
        <p className="footer"> copyright Â© 2018. MyDiary @ Andela. Hammed Noibi.</p>
      </div>
    );
  }
}
EditEntry.propTypes = {
  updateEntryData: PropTypes.object,
  entry: PropTypes.object,
  history: PropTypes.object,
  fetchSingleEntry: PropTypes.func,
  match: PropTypes.any,
  saveInput: PropTypes.func,
  updateSingleEntry: PropTypes.func
};

const mapStateToProps = state => ({
  entry: state.selectedEntry.data || {},
  updateEntryData: state.updateEntryData || {}
});
const mapDispatchToProps = dispatch => ({
  fetchSingleEntry: entryId => dispatch(fetchEntry({ entryId })),
  updateSingleEntry: data => dispatch(updateEntry(data)),
  saveInput: (field, value) => dispatch(saveUpdateInput(field, value)),
  dispatch
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditEntry);
