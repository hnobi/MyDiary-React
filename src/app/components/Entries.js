import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import fetchEntries from '../redux/actions/entries';
import { getAuthToken } from '../services/AuthToken';
import '../assets/css/main.css';

class Entries extends Component {
  componentDidMount() {
    const authToken = getAuthToken();
    const { history } = this.props;
    if (!authToken) {
      history.replace('/login');
    } else {
      const { fetchUserEntries } = this.props;
      fetchUserEntries();
    }
  }

  openEntry(entryId) {
    const { history } = this.props;
    history.push(`/entries/${entryId}`);
  }

  render() {
    const { entries } = this.props;
    console.log('#$%$', entries);
    const noEntries = 'There are no entries available';
    const entriesList = entries.map((item, index) => (
      <tr key={index}>
        <td>{index + 1}</td>
        <td>{item.date}</td>
        <td>{item.title}</td>
        <td>
          {item.entry}
        </td>
        <td>
          <a>
            <button className="view-button" onClick={() => this.openEntry(item.id)}>
              View Entry
            </button>
          </a>
        </td>
      </tr>
    ));
    return (
      <div>
        <nav className="navcolor">
          <div id="logo" className="nav-logo">
            <Link to="/" style={{ display: 'flex' }} className="nav-logo">
              <img src="../assets/img/logo-small2.png" alt="logo" />
              <span> My Diary</span>
            </Link>
          </div>
          <div style={{ margin: 'auto' }} />
          <div className="menu">
            <ul>
              <li>
                <Link to="/entries">View Entries</Link>
              </li>
              <li>
                <Link to="/add-entry">Add Entry</Link>
              </li>
              <li id="logout">
                <Link to="/logout">Logout</Link>
              </li>
            </ul>
          </div>
          <Link to="user_detail.html" id="user">
            <div className="user">
              <img
                src="../assets/img/default-img.png"
                alt="User"
                width="96"
                height="96"
                id="nav-img"
              />
              <span id="nav-name" />
            </div>
          </Link>
        </nav>
        <section>
          <h2 className="heading">My Entries</h2>
          {entries.length === 0 ? (
            noEntries
          ) : (
            <table id="entries-table">
              <thead>
                <tr>
                  <th>Id</th>
                  <th>Date</th>
                  <th>Title</th>
                  <th>Entry</th>
                  <th>Action</th>
                </tr>
              </thead>
              <tbody id="tbody">{entriesList}</tbody>
            </table>
          )}
        </section>
        <div className="push" />
        <p className="footer"> copyright Â© 2018. MyDiary @ Andela. Hammed Noibi.</p>
      </div>
    );
  }
}

Entries.propTypes = {
  entries: PropTypes.array,
  fetchUserEntries: PropTypes.func,
  openSingleEntry: PropTypes.any,
  history: PropTypes.object
};
const mapStateToProps = state => ({
  entries: state.entries.allEntries
});
const mapDispatchToProps = dispatch => ({
  // openSingleEntry: entryId => dispatch({
  //   type: 'SET_ACTIVE_VIEW_ENTRY',
  //   payload: entryId
  // }),
  fetchUserEntries: () => dispatch(fetchEntries())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Entries);
