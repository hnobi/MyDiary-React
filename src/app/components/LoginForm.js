import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { LOGIN_SAVE_INPUT } from '../redux/constant/actionTypes';
import userLogin from '../redux/actions/login';
import Alert from './Alert';
import '../assets/css/main.css';
import '../assets/css/form.css';

export class LoginForm extends Component {
  render() {
    const { saveInput, handleUserLogin, history } = this.props;
    const { username, password } = this.props.loginData.input;
    const loginInput = this.props.loginData.input;
    const { loading, message, status } = this.props.loginData;
    const statusClassName = loading ? 'loading' : '';

    if (status === 'success') {
      setTimeout(() => {
        history.push('/add-entry');
      }, 500);
    }
    return (
      <div>
        <main>
          {loading === false && <Alert message={message} status={status} />}
          <div id="after-click">
            <h3>Invalid username or password</h3>
          </div>
          <h2>Sign in</h2>
          <div className="Sign-in-out">
            <form
              id="signin"
              onSubmit={(e) => {
                e.preventDefault();
                handleUserLogin(loginInput);
              }}
            >
              <label htmlFor="username">Username</label>
              <input
                type="text"
                id="username"
                name="username"
                placeholder="username"
                required
                onChange={e => saveInput(e.target.name, e.target.value)}
                defaultValue={username}
              />

              <label htmlFor="password">Password</label>
              <input
                type="password"
                id="password"
                name="password"
                placeholder="secret password"
                required
                onChange={e => saveInput(e.target.name, e.target.value)}
                defaultValue={password}
              />
              <input type="submit" value="Login" className={statusClassName} />
            </form>
            <p>
              dont have an Account ? <Link to="/signup"> Sign up</Link>{' '}
            </p>
            <div className="push" />
          </div>
          <p className="footer">copyright Â© 2018. MyDiary @ Andela. Hammed Noibi.</p>
        </main>
      </div>
    );
  }
}

LoginForm.propTypes = {
  saveInput: PropTypes.func,
  handleUserLogin: PropTypes.func,
  username: PropTypes.string,
  password: PropTypes.string,
  history: PropTypes.any,
  loginData: PropTypes.object
};

const mapStateToProps = state => ({
  loginData: state.loginData
});
const mapDispatchToProps = dispatch => ({
  saveInput: (field, value) => {
    dispatch({
      type: LOGIN_SAVE_INPUT,
      payload: { field, value }
    });
  },
  handleUserLogin: (data, history) => dispatch(userLogin(data, history))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginForm);
