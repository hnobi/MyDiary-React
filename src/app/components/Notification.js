import React from "react";

import { ToastContainer, ToastStore } from "react-toasts";

class Alert extends React.Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    ToastStore[this.props.status](this.props.message);
  }
  render() {
    return (
      <div>
        <ToastContainer
          store={ToastStore}
          position={ToastContainer.POSITION.TOP_CENTER}
        />
      </div>
    );
  }
}

export default Alert;
