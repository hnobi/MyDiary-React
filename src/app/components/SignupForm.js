import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import userSignup from '../redux/actions/signUp';
import Loader from './spinner';
import '../assets/css/main.css';
import '../assets/css/form.css';

export class SignupForm extends Component {
  constructor() {
    super();
    this.state = {
      fullname: '',
      email: '',
      username: '',
    };
  }

  handleInputchange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleDSignupSubmit = (e) => {
    e.preventDefault();
    this.props.handleUserSignup(this.state);
  }

  render() {
    const { history } = this.props;
    const {
      fullname, username, email, password
    } = this.state;

    const { loading, status } = this.props.signupData;
    const statusClassName = loading ? 'loading' : '';
    if (status === 'success') {
      setTimeout(() => {
        history.push('/login');
      }, 500);
    }
    return (
      <div>
        <nav>
          <div id="logo">
            <Link to="/login">Login</Link>
          </div>
        </nav>
        <main>
          <h2>Sign up</h2>
          <div className="Sign-in-out">
            <form
              id="signup"
              onSubmit={this.handleDSignupSubmit}
            >
              <p>
                Already have an Account ?<Link to="/login">Sign in</Link>
              </p>
              <label htmlFor="fullname">Fullname</label>
              <input
                type="text"
                id="fullname"
                name="fullname"
                placeholder="Your fullname..."
                required
                onChange={this.handleInputchange}
                value={fullname}
              />
              <label htmlFor="email">Email</label>
              <input
                type="email"
                id="email"
                name="email"
                placeholder="Email address..."
                required
                onChange={this.handleInputchange}
                defaultValue={email}
              />

              <label htmlFor="username">Username</label>
              <input
                type="text"
                id="username"
                name="username"
                placeholder=" username..."
                required
                pattern="^[(A-Za-z)|0-9].{1,15}$"
                title="Username may only contain letters or numbers and must be between 2 and 15 characters"
                onChange={this.handleInputchange}
                defaultValue={username}
              />

              <label htmlFor="country">Password</label>
              <input
                type="password"
                id="password"
                name="password"
                placeholder="password"
                pattern="^((?=.*[0-9])(?=.*[a-zA-Z])).{8,12}$"
                title="Password must contain atleast one number and one letter and must be between 8 to 12 characters "
                required
                defaultValue={password}
                onChange={this.handleInputchange}
              />

              <button type="submit" className={statusClassName}>
                {loading ? <Loader color = {'#fff'} size = {30} />
                  : 'Signup'}
              </button>
            </form>
          </div>
          <div className="push" />
        </main>
        <p className="footer">
          {' '}
          copyright Â© 2018. &nbsp;MyDiary @ Andela.&nbsp; By &nbsp;
          <small>Hammed Noibi.</small>
        </p>
      </div>
    );
  }
}
SignupForm.propTypes = {
  handleUserSignup: PropTypes.func,
  signupData: PropTypes.object,
  history: PropTypes.object
};

const mapStateToProps = state => ({
  signupData: state.signupData
});
const mapDispatchToProps = dispatch => ({
  handleUserSignup: data => dispatch(userSignup(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignupForm);
