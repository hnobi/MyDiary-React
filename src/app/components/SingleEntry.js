import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import moment from 'moment';
import fetchEntry from '../redux/actions/viewEntry';
import { getAuthToken } from '../services/AuthToken';
import '../assets/css/main.css';

class SingleEntry extends Component {
  componentDidMount() {
    const authToken = getAuthToken();
    const { history } = this.props;
    if (!authToken) {
      history.replace('/login');
    } else {
      const { fetchSingleEntry, match } = this.props;
      const { entryId } = match.params;
      fetchSingleEntry(entryId);
    }
  }

  render() {
    const { entry } = this.props;
    return (
      <div className="body-image">
        <nav className="navcolor">
          <div id="logo" className="nav-logo">
            <Link to="/" style={{ display: 'flex' }} className="nav-logo">
              <img src="../assets/img/logo-small2.png" alt="logo" />
              <span> My Diary</span>
            </Link>
          </div>
          <div style={{ margin: 'auto' }} />
          <div className="menu">
            <ul>
              <li>
                <Link to="/entries">View Entries</Link>
              </li>
              <li id="logout">
                <Link to="/logout">Logout</Link>
              </li>
            </ul>
          </div>
          <Link to="/profile" id="user">
            <div className="user">
              <img
                src="../assets/img/default-img.png"
                alt="User"
                width="96"
                height="96"
                id="nav-img"
              />
              <span id="nav-name" />
            </div>
          </Link>
        </nav>
        <section>
          <h2 className="heading">Entry information</h2>

          <div className="container view-entry">
            <div className="title">
              <h3>Entry title</h3>
              <p id="title">{entry.title}</p>
            </div>
            <h3>Entry</h3>
            <div className="entry">
              <p id="content">{entry.entry}</p>
            </div>
            <div className="date">
              <h3>Posted : </h3>
              <p id="date">
                {moment(entry.date).format('MMMM Do YYYY')}              </p>
            </div>
            <div
              className="div"
              style={{
                display: 'flex',
                width: '100%',
                justifyContent: 'space-between',
                margin: 'auto'
              }}
            >
              <Link to={`/entries/edit/${entry.id}`}>
                <button className="edit edit-btn">edit</button>
              </Link>
              {/* <a id="delete" onClick={() => deleteSingleEntry(entry.id)}>
                <button className="delete  delete-btn">delete</button>
              </a> */}
            </div>
          </div>
        </section>
        <div className="push" />
        <p className="footer "> copyright Â© 2018. MyDiary @ Andela. Hammed Noibi.</p>
      </div>
    );
  }
}

SingleEntry.propTypes = {
  entries: PropTypes.array,
  selectedEntry: PropTypes.object,
  entry: PropTypes.object,
  history: PropTypes.object,
  fetchSingleEntry: PropTypes.any,
  match: PropTypes.any
};

const mapStateToProps = state => ({
  entries: state.entries,
  selectedEntry: state.selectedEntry || {},
  entry: state.selectedEntry.data || {}
});
const mapDispatchToProps = dispatch => ({
  fetchSingleEntry: entryId => dispatch(fetchEntry({ entryId }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SingleEntry);
