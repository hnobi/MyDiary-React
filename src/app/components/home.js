import React from 'react';
import ReactDOM from 'react-dom';

// import './myStyles.scss';;

const Home = () => {
  return (
    <div className="app">
          <img alt="header" src="/dist/images/a.png" className="app-header" />

      <p>
        We are a most promising species, Mr. Spock, as predators go. Did you know that? I frequently
        have my doubts. I dont. Not any more. And maybe in a thousand years or so, we will be able
        to prove it.
      </p>
      <p>- Captain Kirk</p>
    </div>
  );
};
export default Home