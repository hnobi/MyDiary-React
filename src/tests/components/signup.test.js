import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { SignupForm } from '../../app/components/SignupForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const signupData = {
  input: {
    fullname: 'hammede',
    email: 'ass@jwjjw.com',
    username: 'hnobi',
    password: 'Hnonie23'
  }
};
test('renders Signupnt without crashing', () => {
  const wrapper = shallow(
    <SignupForm
      signupData={signupData}
      handleUserSignup={() => jest.fn()}
      saveInput={() => jest.fn()}
    />
  );
  wrapper
    .find('input')
    .at(0)
    .simulate('change', {
      target: { value: 'fullname' }
    });
  wrapper
    .find('input')
    .at(1)
    .simulate('change', {
      target: { value: 'email' }
    });
  wrapper
    .find('input')
    .at(2)
    .simulate('change', {
      target: { value: 'username' }
    });
  wrapper
    .find('input')
    .at(3)
    .simulate('change', {
      target: { value: 'password' }
    });
  const form = wrapper.find('form');
  form.simulate('submit', {
    preventDefault: jest.fn()
  });
  expect(wrapper).toMatchSnapshot();
});
test('renders SignupForm component without crashing', () => {
  const store = mockStore({ signupData });

  shallow(
    <SignupForm
      store={store}
      signupData={signupData}
      handleUserSignup={() => jest.fn()}
      saveInput={() => jest.fn()}
    />
  );
});
