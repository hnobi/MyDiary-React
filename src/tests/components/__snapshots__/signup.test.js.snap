// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Signupnt without crashing 1`] = `
<div>
  <nav>
    <div
      id="logo"
    >
      <Link
        replace={false}
        to="/login"
      >
        Login
      </Link>
    </div>
  </nav>
  <main>
    <h2>
      Sign up
    </h2>
    <div
      className="Sign-in-out"
    >
      <form
        id="signup"
        onSubmit={[Function]}
      >
        <p>
          Already have an Account ?
          <Link
            replace={false}
            to="/login"
          >
            Sign in
          </Link>
        </p>
        <label
          htmlFor="fullname"
        >
          Fullname
        </label>
        <input
          defaultValue="hammede"
          id="fullname"
          name="fullname"
          onChange={[Function]}
          placeholder="Your fullname..."
          required={true}
          type="text"
        />
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          defaultValue="ass@jwjjw.com"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email address..."
          required={true}
          type="email"
        />
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          defaultValue="hnobi"
          id="username"
          name="username"
          onChange={[Function]}
          pattern="^[(A-Za-z)|0-9].{1,15}$"
          placeholder=" username..."
          required={true}
          title="Username may only contain letters or numbers and must be between 2 and 15 characters"
          type="text"
        />
        <label
          htmlFor="country"
        >
          Password
        </label>
        <input
          defaultValue="Hnonie23"
          id="password"
          name="password"
          onChange={[Function]}
          pattern="^((?=.*[0-9])(?=.*[a-zA-Z])).{8,12}$"
          placeholder="password"
          required={true}
          title="Password must contain atleast one number and one letter and must be between 8 to 12 characters "
          type="password"
        />
        <input
          className=""
          type="submit"
          value="Sign up"
        />
      </form>
    </div>
    <div
      className="push"
    />
  </main>
  <p
    className="footer"
  >
     
    copyright © 2018.  MyDiary @ Andela.  By  
    <small>
      Hammed Noibi.
    </small>
  </p>
</div>
`;
